src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1001,6 +1010,191 @@ else if (systemUuid != null) {
+                .setParameter("certIds", caCertIds)
+                .getResultList());
+        }
+
+        if (idCertIds != null && !idCertIds.isEmpty()) {
+            String idCertSql =
+                "SELECT idcert.serial_id AS serialIds " +
+                "FROM cp_id_cert idcert " +
+                "WHERE idcert.id IN (:certIds)";
+
+            serialIds.addAll(this.currentSession()
+                .createSQLQuery(idCertSql)
+                .setParameter("certIds", idCertIds)
+                .getResultList());
+        }
+
+        return serialIds.stream().map(BigInteger::longValue).collect(Collectors.toList());
+    }
+
+    /**
+     * Deletes {@link Consumer}s based on the provided consumer ids.
+     *
+     * @param consumerIds - ids of the consumers to delete.
+     * @return the number of consumer that were deleted.
+     */
+    public int deleteConsumers(Collection<String> consumerIds) {
+        if (consumerIds == null || consumerIds.isEmpty()) {
+            return 0;
+        }
+
+        String deleteConsumersStatement = "DELETE FROM cp_consumer WHERE id IN (:consumerIds)";
+        return this.currentSession()
+            .createSQLQuery(deleteConsumersStatement)
+            .setParameter("consumerIds", consumerIds)
+            .executeUpdate();
+    }
+
+    /**
+     * Retrieves the Ids for inactive {@link Consumer}s based on the provided last checked in retention
+     * date and the last updated retention date. Consumers are considered inactive if the have a checked
+     * in date and that date is older than the provided checked in retention date, or the consumer's update
+     * date is older than the provided last update retention date. Also, the consumer must not have any
+     * attached entitlements and must have a non-manifest type to be considered inactive.
+     *
+     * @param lastCheckedInRetention
+     *  - consumers that have not checked in before this date are considered inactive.
+     * @param lastUpdatedRetention
+     *  - if the consumer has no checked in date, then the consumers that have an update date older than
+     *    the provided retention date is considered inactive.
+     * @return a list of Ids for inactive {@link Consumer}s.
+     */
+    @Transactional
+    public List<String> getInactiveConsumerIds(Instant lastCheckedInRetention, Instant lastUpdatedRetention) {
+        if (lastCheckedInRetention == null) {
+            throw new IllegalArgumentException("Last checked-in retention date cannot be null.");
+        }
+
+        if (lastUpdatedRetention == null) {
+            throw new IllegalArgumentException("Last updated retention date cannot be null.");
+        }
+
+        String sql =
+            "SELECT consumer.id " +
+            "FROM cp_consumer consumer " +
+            "JOIN cp_consumer_type type ON type.id=consumer.type_id " +
+            "LEFT JOIN " +
+            "    (SELECT DISTINCT consumer_id FROM cp_entitlement) ent ON ent.consumer_id=consumer.id " +
```

Comment:
Also, I don't think the subquery here will work for the purpose of fetching only consumers without entitlements.
it should probably just be `...LEFT JOIN cp_entitlement ent ON ent.consumer_id=consumer.id...`

Reply:
Good catch! Updated